lib/rubocop/cop/style/frozen_string_literal_comment.rb

```diff
@@ -122,41 +122,22 @@ def remove_comment(corrector, node)
 
         def insert_comment(corrector)
```

Comment:
Since the value of `line` here depends on whether `last_special_comment` is `nil` or not, I think it'd be easier to follow what's going on if that was lifted into the conditional, i.e.:

```ruby
last_special_comment = last_special_comment(processed_source)

if last_special_comment.nil?
  line_range = processed_source.buffer.line_range(1)

  corrector.insert_before(line_range, preceding_comment)
else
  line_range = processed_source.buffer.line_range(last_special_comment.line)

  corrector.insert_after(line_range, following_comment)
end
```

I would personally rather have this piece of duplication and make the conditional cases clearer. WDYT?

Reply:
You're right: That piece of code can be improved. I just pushed a variant of your second suggestion.